### DISCUSSION TITLE
How does this compare with elastic search-ui?

### DISCUSSION THREAD

**Discussion by tombburnell** [appbaseio/reactivesearch#1866] [Q&A] [answered]
What's the difference between them?
Is there a feature comparison somewhere?
Anyone have experiences of using both?

Not sure why I should choose one over the other..

**Comment by siddharthlatest** [comment, marked as answer]
The primary differences are:

**DSL support:** ReactiveSearch supports the entire Elasticsearch query DSL, as we started out with this goal. Since the Elasticsearch/OpenSearch divergence, both search engines are supported by ReactiveSearch. Whereas, Elastic search UI natively supports Elastic App Search, a refined set of REST APIs built on top of Elasticsearch. They have a headless version (no UI) which provides the primitives for a user to configure it with another search engine.

Practically, if you want more control over your Elasticsearch queries, this is natively supported within ReactiveSearch (e.g. use nested types, joins, or write your own bool clause, or customize the aggregations) so you never hit a wall.

**Note:** ReactiveSearch v3 onwards recommends the use of [ReactiveSearch API](https://github.com/appbaseio/reactivesearch-api) server (available as open-source Apache 2.0 licensed software) to be run with an Elasticsearch or OpenSearch cluster.

**All batteries included:** Another difference is that ReactiveSearch's approach is to be all batteries included, it provides over [20 UI components for React](https://docs.appbase.io/docs/reactivesearch/v3/overview/components/) including geo search related components (maps), and Vue version has over [10 UI components](https://docs.appbase.io/docs/reactivesearch/vue/overview/Components/). Elastic Search UI today is available for React and has [9 UI components](https://github.com/elastic/search-ui#4-components).

**Frontend frameworks supported:** ReactiveSearch is available for React and Vue. Searchbox (another suite of UI libraries maintained by appbase.io) is available for React, Vue, Vanilla JS (headless version), React Native and Flutter. Elastic search UI is available for React and for Vanilla JS (headless version),

**Reply by tombburnell** [reply]
Thanks for the explanation. I'd like to try it out but I hit this issue today: https://github.com/appbaseio/reactivesearch-api/issues/251 and couldn't get the docker instance started as it was complaining about not finding license key.
